### PR TITLE
Move tutorials

### PR DESCRIPTION
**Pull request by wxtim**
These changes partially address https://github.com/cylc/cylc-doc/issues/40

sibling of https://github.com/cylc/cylc-doc/pull/371

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Does not need tests (This is itself a test, moved from Cylc Doc).
- [x] Appropriate change log entry included.
- [x] No documentation update required - this is part of ongoing work at https://github.com/cylc/cylc-doc/issues/40 and will require follow up.

### CODE REVIEW

File: .github/actions/install-dependencies/action.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Install dependencies
```
Comment (MetRonnie) [inline .github/actions/install-dependencies/action.yml:1]: No need to copy this action here, can just use like so
```yaml
- name: Install docs/tutorial dependencies
  uses: cylc/cylc-doc/.github/actions/install-dependencies@master
```

File: .github/workflows/test_tutorial_workflow.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: configure python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+
+      - name: checkout cylc-doc
+        uses: actions/checkout@v2
```
Comment (MetRonnie) [inline .github/workflows/test_tutorial_workflow.yml:30]:
```suggestion
      - name: checkout
        uses: actions/checkout@v2
```

File: .github/workflows/test_tutorial_workflow.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+      - name: install dependencies
+        uses: ./.github/actions/install-dependencies
```
Comment (MetRonnie) [inline .github/workflows/test_tutorial_workflow.yml:33]: See my earlier comment
```suggestion
      - name: Install docs/tutorial dependencies
        uses: cylc/cylc-doc/.github/actions/install-dependencies@master
```

File: .github/workflows/test_tutorial_workflow.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+      - name: install cylc-doc
```
Comment (MetRonnie) [inline .github/workflows/test_tutorial_workflow.yml:35]:
```suggestion
      - name: install cylc-flow
```

**Comment by wxtim** [comment]
Test failures look like those fixed in #4574

**Review by MetRonnie — APPROVED** [review]

**Review by hjoliver — CHANGES_REQUESTED** [review]
Can `bin/run-validate` be a) renamed to makes its purpose clear, e.g. `run-validate-tutorials`; and b) moved to `etc/bin/`where the other developer scripts live? (it seems a weird thing to have as the sole resident of a top-level bin directory in the cylc-flow repo)

File: etc/bin/run-validate-tutorials [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+set -eu
+cd "$(dirname "$0")"
+
+for FILE in $(echo ../cylc/flow/etc/tutorial/*/.validate) ; do
```
Comment (MetRonnie) [inline etc/bin/run-validate-tutorials:22]: This path looks like it needs updating
```
running ../cylc/flow/etc/tutorial/*/.validate
etc/bin/run-validate-tutorials: line 25: cd: ../cylc/flow/etc/tutorial/*: No such file or directory
```
https://github.com/cylc/cylc-flow/runs/4799870817?check_suite_focus=true#step:7:8

**Review by MetRonnie — CHANGES_REQUESTED** [review]
Also changelog in conflict at the moment

**Review by MetRonnie — APPROVED** [review]

**Review by hjoliver — APPROVED** [review]

**Pull request merged**